### DISCUSSION TITLE
enable using input as canny map

### DISCUSSION THREAD

**Discussion by tariks** [filipstrand/mflux#549] [Ideas & Suggestions: Features]
My appreciation for great work. mflux running significantly faster than comfy sdxl on my m1 macbook. 

It would be a great enhancement to enable input of an already preprocessed control image. In cases where user already has a black and white line image representing edges / scribbles, running canny over it often is a downgrade in quality. 

Said another way, running edge detection on detected edges makes things worse.

**Comment by filipstrand** [comment]
Thank you very much! Yes that is a nice feature and something I have thought of myself too. 

Another feature improvement related to the edge detection is more user control over how the input image gets converted to the canny representation. Currently there are some hardcoded values there that can impact generation a bit, so would be nice to also improve that user experience a bit by letting the user control those values
